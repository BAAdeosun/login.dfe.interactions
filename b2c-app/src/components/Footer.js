import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
