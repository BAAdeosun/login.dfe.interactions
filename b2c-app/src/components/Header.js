import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;