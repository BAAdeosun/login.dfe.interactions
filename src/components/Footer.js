import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer