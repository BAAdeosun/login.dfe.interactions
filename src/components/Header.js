import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header