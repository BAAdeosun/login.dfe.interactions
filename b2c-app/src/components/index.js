import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}