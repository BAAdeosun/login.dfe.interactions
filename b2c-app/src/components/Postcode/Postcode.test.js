import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

const testErrors = {
    postcode: {
        text: 'this is a test'
    }
};
const updateParentMock = jest.fn();

it('renders without crashing', () => {
    shallow(<components.Postcode />);
});

it('renders correctly without props passed in', () => {
    const tree = renderer
        .create(<components.Postcode />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with errors', () => {
    const tree = renderer.create(
        <components.Postcode showErrors={true} visibleErrors={testErrors} />);

    expect(tree.toJSON()).toMatchSnapshot()
});

it('calls validation, sets errors and calls onChange callback', () => {

    const mockOnChangeCallback = jest.fn();

    const wrapper = mount(
        <components.Postcode
            showErrors={true}
            onChange={mockOnChangeCallback}
            visibleErrors={testErrors}
            updateParentErrors={updateParentMock} />
    );

    const postcodeInput = wrapper.find('#postcodeCustom');
    const validationSpy = jest.spyOn(wrapper.instance(), 'isValidPostcode');
    let changeEvent = {
        target:
        {
            name: 'postcode',
            value: 'RG02 1AB'
        }
    };

    //default values set
    expect(wrapper.state().postcode).toBe(null);
    expect(wrapper.state().errors.postcode.text).toEqual('Enter your postcode');

    //simulate entering valid postcode
    postcodeInput.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().postcode).toEqual('RG02 1AB');
    //check error message is empty
    expect(wrapper.state().errors.postcode.text).toEqual('');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ postcode: 'RG02 1AB' });

    //simulate entering valid postcode without space
    changeEvent.target.value = 'RG021AB';
    postcodeInput.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated and the space is added to the postcode
    expect(wrapper.state().postcode).toEqual('RG02 1AB');
    //check error message is empty
    expect(wrapper.state().errors.postcode.text).toEqual('');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ postcode: 'RG02 1AB' });

    //simulate entering invalid postcode (no space in it should make it fail validation)
    changeEvent.target.value = 'invalid';
    postcodeInput.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().postcode).toEqual('invalid');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ postcode: null });
    //check error message is set
    expect(wrapper.state().errors.postcode.text).toEqual('Enter a valid postcode');

    //simulate entering empty postcode
    changeEvent.target.value = '';
    postcodeInput.simulate('change', changeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().postcode).toEqual('');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ postcode: null });
    //check error message is set
    expect(wrapper.state().errors.postcode.text).toEqual('Enter your postcode');

});