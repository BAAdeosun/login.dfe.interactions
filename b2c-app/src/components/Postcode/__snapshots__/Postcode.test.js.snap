// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with errors 1`] = `
<div
  className="govuk-form-group govuk-form-group--error"
>
  <label
    className="govuk-label"
    htmlFor="postcodeCustom"
  >
    Enter your postcode
  </label>
  <span
    className="govuk-error-message"
    id="postcodeError"
  >
    <span
      className="govuk-visually-hidden"
    >
      Error:
    </span>
    this is a test
  </span>
  <span
    className="govuk-hint"
    id="postcode-hint"
  >
    For example, SW1A 1AA
  </span>
  <input
    className="govuk-input govuk-input--width-10"
    id="postcodeCustom"
    name="postcode"
    noValidate={true}
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`renders correctly without props passed in 1`] = `
<div
  className="govuk-form-group "
>
  <label
    className="govuk-label"
    htmlFor="postcodeCustom"
  >
    Enter your postcode
  </label>
  <span
    className="govuk-hint"
    id="postcode-hint"
  >
    For example, SW1A 1AA
  </span>
  <input
    className="govuk-input govuk-input--width-10"
    id="postcodeCustom"
    name="postcode"
    noValidate={true}
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
